### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Daniil-Pankieiev**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,58 @@
...
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    @classmethod
+    def checkvalue(cls, other: "Distance") -> float:
```
Comment (LiudmylaKulinchenko) [inline app/main.py:12]: You can just use type annotations and don't check for other types. It is up to developer call methods with the correct data types if there are annotations
So you can remove this method

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]
Fix type annotations in all methods according to my comment

**Review by Familenko — APPROVED** [review]

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,55 @@
...
+
+    def __le__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km <= other.km
+        return self.km <= other
+
+    def __ge__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km >= other.km
+        return self.km >= other
```
Comment (fsocie7y) [inline app/main.py:55]: try to rewrite these two methods in one line each. Use previously created magic methods
Hint: <= is the same as not >

**Review by fsocie7y — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,51 @@
...
+    def __eq__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km == other.km
+        return self.km == other
+
+    def __le__(self, other: Distance | int | float) -> bool:
+        return not self.__gt__(other)
+
+    def __ge__(self, other: Distance | int | float) -> bool:
+        return not self.__lt__(other)
```
Comment (fsocie7y) [inline app/main.py:51]: cool, but one more point
You can call these magic methods using operands
ex: `<` instead `self.__lt__(other)`

**Review by fsocie7y — CHANGES_REQUESTED** [review]

**Review by fsocie7y — APPROVED** [review]
well done ^_^

**Pull request closed without merging**